apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -50,16 +57,38 @@ func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
 	return errs
 }
 
-func validateRAID(r *RAIDConfig) error {
-	if r == nil {
-		return nil
+func validateBMCAccess(s BareMetalHostSpec, bmcAccess bmc.AccessDetails) []error {
+	var errs []error
+
+	if bmcAccess == nil {
+		return errs
 	}
 
-	if len(r.HardwareRAIDVolumes) > 0 && len(r.SoftwareRAIDVolumes) > 0 {
-		return fmt.Errorf("hardwareRAIDVolumes and softwareRAIDVolumes can not be set at the same time")
+	if s.RAID != nil && len(s.RAID.HardwareRAIDVolumes) > 0 {
+		if bmcAccess.RAIDInterface() == "no-raid" {
+			errs = append(errs, fmt.Errorf("BMC driver %s does not support configuring RAID", bmcAccess.Type()))
+		}
+
+		if len(s.RAID.SoftwareRAIDVolumes) > 0 {
```

Comment:
This isn't really BMC-related, I think I'd prefer to have kept the validateRAID() function as well (this won't be the last RAID validation we need), even if it means repeating the if statement from line 67 in two places.

Reply:
Done, PTAL.